staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go

```diff
@@ -29,22 +29,115 @@ type AdmissionConfiguration struct {
 
 	// Plugins allows specifying a configuration per admission control plugin.
 	// +optional
-	Plugins []AdmissionPluginConfiguration `json:"plugins"`
+	Plugins []AdmissionPluginConfiguration `json:"plugins" protobuf:"bytes,1,rep,name=plugins"`
 }
 
 // AdmissionPluginConfiguration provides the configuration for a single plug-in.
 type AdmissionPluginConfiguration struct {
 	// Name is the name of the admission controller.
 	// It must match the registered admission plugin name.
-	Name string `json:"name"`
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
 
 	// Path is the path to a configuration file that contains the plugin's
 	// configuration
 	// +optional
-	Path string `json:"path"`
+	Path string `json:"path" protobuf:"bytes,2,opt,name=path"`
 
 	// Configuration is an embedded configuration object to be used as the plugin's
 	// configuration. If present, it will be used instead of the path to the configuration file.
 	// +optional
-	Configuration *runtime.Unknown `json:"configuration"`
+	Configuration *runtime.Unknown `json:"configuration" protobuf:"bytes,3,opt,name=configuration"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// +k8s:openapi-gen=true
+// Webhook describes a webhook and the resources and operations it applies to.
+type Webhook struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// The name of the webhook.
+	// Required.
+	Name string `json:"name" protobuf:"bytes,2,opt,name=name"`
+
+	// ClientConfig defines how to communicate with the hook.
+	// Required
+	ClientConfig WebhookClientConfig `json:"clientConfig" protobuf:"bytes,3,opt,name=clientConfig"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// +k8s:openapi-gen=true
+// WebhookClientConfig contains the information to make a connection with the webhook
+type WebhookClientConfig struct {
```

Comment:
Copied from the admission webhook configuration?

Reply:
yeah, it essentially needs the same methods but there were incompatibilities with the admission implementation. Talking internally we decided to factor it into a common package. This PR doesn't look to refactor the admission pieces but does allow the owners of that package the ability to go that route in the future.